CRM/Contact/Form/Merge.php

```diff
@@ -208,7 +208,8 @@ public function preProcess() {
     $this->_contactType = $main['contact_type'];
     $this->addElement('checkbox', 'toggleSelect', NULL, NULL, array('class' => 'select-rows'));
 
-    $this->assign('mainLocBlock', json_encode($rowsElementsAndInfo['main_loc_block']));
+    $this->assign('mainLocBlock', json_encode($rowsElementsAndInfo['main_details']['location_blocks']));
```

Comment:
I think this line is no longer required. I believe you have removed the only reference to mainLocBlock in the tpl